### PR TITLE
Fix databinding error in Salesforce query

### PR DESCRIPTION
**Pull request by aashikam**
# Description
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6727

### CODE REVIEW

File: ballerina/stream_implementer.bal [hunk 1]
```diff
@@ -57,9 +62,20 @@ class SOQLQueryResultStream {
         self.nextRecordsUrl = response.hasKey(NEXT_RECORDS_URL) ? check response.get(NEXT_RECORDS_URL).ensureType() :
             EMPTY_STRING;
 
-        record{}[] array = response.records;
-        return array;
+        record {}[]|error returnData = self.recordConverter(response.records, self.returnType);
```
Comment (niveathika) [inline ballerina/stream_implementer.bal:65]: Do we need to jump to java side for this? Cant we use cloneWithtype()?

Comment (aashikam) [reply ballerina/stream_implementer.bal:65]: Can we pass a typedesc to cloneWithtype()? ok i'll add that her ethen

File: changelog.md [hunk 2]
```diff
@@ -3,6 +3,10 @@ This file contains all the notable changes done to the Ballerina Salesforce pack
 
 ## [Unreleased]
```
Comment (niveathika) [inline changelog.md:4]: Need to add the changes for this minor release

File: ballerina/Dependencies.toml [hunk 3]
```diff
@@ -93,7 +93,7 @@ modules = [
 [[package]]
 org = "ballerina"
 name = "io"
-version = "1.6.0"
+version = "1.6.1"
```
Comment (gimantha) [inline ballerina/Dependencies.toml:96]: Shouldnt this be a minor release version rather than a patch version? @niveathika @aashikam

Comment (aashikam) [reply ballerina/Dependencies.toml:96]: Yes yes

**Review by gimantha — DISMISSED** [review]
LGTM

File: ballerina/Dependencies.toml (hunk 3 above)
Comment (aashikam) [reply ballerina/Dependencies.toml:96]: This is io package dependency!

Comment (aashikam) [reply ballerina/Dependencies.toml:96]: https://github.com/ballerina-platform/module-ballerinax-salesforce/blob/ca9a5787ce72700e78db7cfa43efc742ed164f2b/gradle.properties#L3 

Salesforce new version: 8.1.0 
Last version in central: 8.0.2

**Review by niveathika — APPROVED** [review]

**Pull request merged**